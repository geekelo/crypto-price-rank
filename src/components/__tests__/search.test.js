import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Search from '../search';

// Mock the useSelector function
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

test('Search component matches snapshot', () => {
  // Mock the useSelector hook
  useSelector.mockReturnValue({ theme: { value: true } });

  const mockHandleFilteredNames = jest.fn();

  const { asFragment } = render(<Search handleFliterednames={mockHandleFilteredNames} />);
  
  // Compare the rendered component to a stored snapshot
  expect(asFragment()).toMatchSnapshot();
});
