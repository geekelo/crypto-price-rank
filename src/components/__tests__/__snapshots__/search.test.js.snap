// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form
      class="searchContain lightsearchContain"
    >
      <div
        class="input lightinput"
      >
        <input
          class="formField lightformField"
          placeholder="Search"
          type="search"
          value=""
        />
      </div>
      <div
        class="input lightinput"
      >
        <select
          class="formField lightformField"
        >
          <option
            value="null"
          >
            - Select Ranking -
          </option>
          <option
            value="priceUsd"
          >
            Rank by Price
          </option>
          <option
            value="supply"
          >
            Rank by Supply
          </option>
        </select>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
